cf-agent/verify_files_utils.c

```diff
@@ -2617,8 +2617,27 @@ static PromiseResult VerifyFileAttributes(EvalContext *ctx, const char *file, co
             }
             else
             {
-                RecordChange(ctx, pp, attr, "Object '%s' had permissions %04jo, changed it to %04jo",
-                             file, (uintmax_t)dstat->st_mode & 07777, (uintmax_t)newperm & 07777);
+                char *object = (S_ISREG(dstat->st_mode)) ? "Regular file"
+                             : (S_ISDIR(dstat->st_mode)) ? "Directory"
+                             : (S_ISSOCK(dstat->st_mode)) ? "Socket"
+                             : (S_ISFIFO(dstat->st_mode)) ? "Named pipe"
+                             : (S_ISCHR(dstat->st_mode)) ? "Block device"
+                             : (S_ISBLK(dstat->st_mode)) ? "Character device"
```

Comment:
Looks like you can change this to a swtich (by doing bitwise AND on some mask);

https://man7.org/tlpi/code/online/dist/files/t_stat.c.html